actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchNotificationActor.java

```diff
@@ -0,0 +1,361 @@
+    if (userIds != null) {
+      participantList = getUsersFromDB(userIds);
+    }
+    if (mentorList != null) {
+      processUserDataAndSendMail(mentorList, courseBatch, operationType, JsonKey.MENTOR);
+    }
+    if (participantList != null) {
+      processUserDataAndSendMail(participantList, courseBatch, operationType, JsonKey.PARTICIPANT);
+    }
+  }
+
+  /*
+   * This method call sendMail method with all the required data
+   *
+   * @param CourseBatch object for course related data
+   *
+   * @param List<Map<String,Object>> list of user
+   *
+   * @param operationType String for specifying the operation type ie: add/remove
+   */
+  @SuppressWarnings("unchecked")
+  private void processUserDataAndSendMail(
+      List<Map<String, Object>> userList,
+      CourseBatch courseBatch,
+      String operationType,
+      String userType) {
+
+    for (Map<String, Object> user : userList) {
+      Map<String, Object> requestMap = new HashMap<String, Object>();
+      requestMap = this.createRequestMap(courseBatch);
+      Map<String, String> userData = null;
+      if (operationType.equals(JsonKey.ADD)) {
+        userData = getUserData(user, JsonKey.ADD);
+        if (userData != null) {
+          if (userType.equals(JsonKey.PARTICIPANT)) {
+            userData.put(JsonKey.EMAIL_TEMPLATE_TYPE, JsonKey.BATCH_LEARNER_ENROL);
+          } else {
+            userData.put(JsonKey.EMAIL_TEMPLATE_TYPE, JsonKey.BATCH_MENTOR_ENROL);
+          }
+        } else {
+          ProjectLogger.log(
+              "BatchOperationNotifierActor: processUserDataAndSendMail : User data is NULL",
+              LoggerEnum.ERROR.name());
+        }
+      } else {
+        userData = getUserData(user, JsonKey.REMOVE);
+        if (userData != null) {
+          if (userType.equals(JsonKey.PARTICIPANT)) {
+            userData.put(JsonKey.EMAIL_TEMPLATE_TYPE, JsonKey.BATCH_LEARNER_UNENROL);
+          } else {
+            userData.put(JsonKey.EMAIL_TEMPLATE_TYPE, JsonKey.BATCH_MENTOR_UNENROL);
+          }
+        } else {
+          ProjectLogger.log(
+              "BatchOperationNotifierActor: processUserDataAndSendMail : User data is NULL",
+              LoggerEnum.ERROR.name());
+        }
+      }
+      if (user.get(JsonKey.USER_ID) != null
+          && StringUtils.isNotBlank((String) user.get(JsonKey.USER_ID))) {
+        requestMap.put(JsonKey.RECIPIENT_USERIDS, (String) user.get(JsonKey.USER_ID));
+      }
+      sendMail(userData, requestMap);
+    }
+  }
+
+  /*
+   * This method takes two courseBatch object as parameter and process update
+   * batch related operation notifications
+   *
+   * @param courseBatchPrev course and email related data before update
+   *
+   * @param courseBatchNew course and email related data after update
+   */
+  private void batchUpdateOperationNotifier(
+      CourseBatch courseBatch, Map<String, Object> mentorParticipantMap) {
+
+    if (mentorParticipantMap.get(JsonKey.REMOVED_MENTORS) != null
```

Comment:
This method is full of code repetition. If list of user IDs are passed in REMOVED_MENTORS, ADDED_MENTORS, REMOVED_PARTICIPANTS, ADDED_PARTICIPANTS then a helper method can be called to notify with different lists and operations to avoid code duplication.